src/Ser/SerCsv.jl

```diff
@@ -17,6 +26,144 @@ function wrap_value(s::AbstractString)
     return s
 end
 
+const CSV_NULL = ""
+
+value_to_string(::Nothing) = CSV_NULL
+value_to_string(::Missing) = CSV_NULL
+value_to_string(x::Any) = wrap_value(string(x))
+
+issimple(::Type{<:Any})::Bool = false
+issimple(::Type{<:Enum})::Bool = true
+issimple(::Type{<:Symbol})::Bool = true
+issimple(::Type{<:Number})::Bool = true
+issimple(::Type{<:AbstractChar})::Bool = true
+issimple(::Type{<:AbstractString})::Bool = true
+issimple(::Type{<:Dates.TimeType})::Bool = true
+
+isnull(::Type{Missing}) = true
+isnull(::Type{Nothing}) = true
+isnull(::Type{<:Any}) = false
+
+isunion(::Type{T}) where {T} = T isa Union
+
+iscomposite(::Type{T}) where {T} = !(issimple(T) || isnull(T) || isunion(T))
+
+function is_valid_union(::Type{T})::Bool where {T}
+    has_simple = false
+    has_composite = false
+    for type in Base.uniontypes(T)
+        if issimple(type)
+            has_simple = true
+        elseif iscomposite(type)
+            has_composite = true
+        end
+    end
+    return has_simple ⊻ has_composite
+end
+
+function could_flatten(::Type{T})::Bool where {T}
+    return if issimple(T)
+        false
+    elseif isnull(T)
+        false
+    elseif iscomposite(T)
+        true
+    elseif isunion(T)
+        is_valid_union(T) && !isnothing(extract_composite(T))
+    else
+        false
+    end
+end
+
+function extract_composite(::Type{T}) where {T}
+    return if iscomposite(T)
+        T
+    elseif isunion(T)
+        types = Base.uniontypes(T)
+        index = findfirst(iscomposite, types)
+        isnothing(index) ? nothing : types[index]
+    else
+        throw(CsvSerializationError("Parameter of type $T must be a Union type or Composite Type."))
+    end
+end
+
+function null_count(::Type{T}) where {T}
+    count = 0
+    for type in fieldtypes(T)
+        count += could_flatten(type) ?  null_count(extract_composite(type)) : 1
+    end
+    return count
+end
+
+function row_values(buf::IOBuffer, data::T; delimiter::String = ",") where {T}
+    count = 1
+    for (name, type) in zip(fieldnames(T), fieldtypes(T))
+        if isunion(type) && !is_valid_union(type)
+            throw(CsvSerializationError("Unsupported Union Type in field $name::$type of type $T."))
+        end
+        if count != 1
+            print(buf, delimiter)
+        end
+        value = getproperty(data, name)
+        if could_flatten(type)
+            if isnothing(value) || ismissing(value)
+                print(buf, repeat(delimiter, null_count(extract_composite(type)) - 1))
```

Comment:
Isn't there a `CSV_NULL` option?

Reply:
I think `CSV_NULL` can be ignored, just repeat the `delimiter` here, because `CSV_NULL` is a string with a length of 0, and it makes no sense to concatenate `CSV_NULL` and `delimiter`